packages/chevrotain/src/parse/grammar/interpreter.ts

```diff
@@ -97,8 +97,8 @@ export abstract class AbstractNextPossibleTokensWalker extends RestWalker {
       this.nextProductionOccurrence = 0
       this.isAtEndOfPath = true
     } else {
-      this.nextProductionName = this.ruleStack.pop()
-      this.nextProductionOccurrence = this.occurrenceStack.pop()
+      this.nextProductionName = this.ruleStack.pop()!
```

Comment:
would a cast to a **non-empty** array type better convey the semantic knowledge that these stacks
are never empty (once parser rule has started)?
- https://stackoverflow.com/questions/56006111/is-it-possible-to-define-a-non-empty-array-type-in-typescript

Reply:
.pop() currently doesn't have special case for non-empty arrays. There's probably a way to get that to work, but this is the simpler, idiomatic way to do this.